weavertest/deployer.go

```diff
@@ -296,39 +298,52 @@ func (d *deployer) StartComponent(req *protos.ComponentToStart) error {
 
 // startGroup starts the provided co-location group in a subprocess, if it
 // hasn't already been started.
-//
-// REQUIRES: d.mu is held.
-func (d *deployer) startGroup(group *group) error {
+func (d *deployer) startGroup(g *group) error {
+	started := func() bool {
+		g.mu.Lock()
+		defer g.mu.Unlock()
+		return len(g.envelopes) > 0
+	}
+
+	if started() {
```

Comment:
I think this might be susceptible to races? Can't two goroutines both see `len(g.envelopes) == 0` and then both start subprocesses?

Reply:
Ugh. Done!